x-pack/plugins/task_manager/server/queries/mark_available_tasks_as_claimed.ts

```diff
@@ -166,12 +149,42 @@ function getSortByPriority(definitions: TaskTypeDictionary): estypes.SortCombina
   };
 }
 
+// getClaimSort() is used to generate sort bits for the ES query
+// should align with claimSort() below
 export function getClaimSort(definitions: TaskTypeDictionary): estypes.SortCombinations[] {
   const sortByPriority = getSortByPriority(definitions);
   if (!sortByPriority) return [SortByRunAtAndRetryAt];
   return [sortByPriority, SortByRunAtAndRetryAt];
 }
 
+// claimSort() is used to sort tasks returned from a claimer
+// should align with getClaimSort() above
+export function claimSort(
```

Comment:
can we add a unit test for this?

Reply:
Added an iniital set of tests in comment 450acb39c6cb6060bc38ba48f39b0fae698a3e74  - and then thought about some more I can add - sorting both priority and date to make sure we have the order right.

Not sure if I mentioned this, but the tests were broken when first written (as you would expect), but were failing in a really hard-to-determine fashion.  You'll never guess: `TaskTypeDictionary::get()` being passed a bad type name! :-), throwing an error. I'm going to open an issue to remove that `throw` ...